avm/res/sql/instance-pool/tests/e2e/waf-aligned/main.test.bicep

```diff
@@ -0,0 +1,63 @@
+targetScope = 'subscription'
+
+metadata name = 'Using only defaults'
+metadata description = 'This instance deploys the module with the minimum set of required parameters.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-sql.instancepool-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param resourceLocation string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+// e.g., for a module 'network/private-endpoint' you could use 'npe' as a prefix and then 'waf' as a suffix for the waf-aligned test
+param serviceShort string = 'sipwaf'
+
+@description('Optional. A token to inject into the name of each resource. This value can be automatically injected by the CI.')
+param namePrefix string = '#_namePrefix_#'
+
+var instancePoolName = '${namePrefix}${serviceShort}001'
```

Comment:
To keep the names all at the same location, we usually would recommend to not have a variable here, but specify it in line 44, and inside the `dependencies.bicep` file return the name from the dependency as an output that we reference in the test deployment. This also creates an implicit dependency. Just a suggestion, so if you don't agree, feel free to resolve.

Reply:
same feedback for the defaults test case

Reply:
This is not getting created as part of the dependencies, the only reason I created a variable is because I need it in the dependencies file for naming convention. If that is the preferred approach I am happy to comply

Reply:
Ah I see, it's just used in the delegation. Ah well, I didn't see that it must be in the delegation's name. Thanks for updating though 💪